packages/arb-token-bridge-ui/src/util/withdrawals/fetchTokenWithdrawalsFromEventLogsSequentially.ts

```diff
@@ -0,0 +1,147 @@
+import { constants } from 'ethers'
+import { Provider, BlockTag } from '@ethersproject/providers'
+import { Erc20Bridger, getArbitrumNetwork } from '@arbitrum/sdk'
+
+import {
+  fetchTokenWithdrawalsFromEventLogs,
+  FetchTokenWithdrawalsFromEventLogsParams
+} from './fetchTokenWithdrawalsFromEventLogs'
+import { getNonce } from '../AddressUtils'
+import { fetchL2Gateways } from '../fetchL2Gateways'
+import { wait } from '../ExponentialBackoffUtils'
+
+async function getGateways(provider: Provider): Promise<{
+  standardGateway: string
+  wethGateway: string
+  customGateway: string
+  otherGateways: string[]
+}> {
+  const network = await getArbitrumNetwork(provider)
+
+  const standardGateway = network.tokenBridge?.childErc20Gateway
+  const customGateway = network.tokenBridge?.childCustomGateway
+  const wethGateway = network.tokenBridge?.childWethGateway
+  const otherGateways = await fetchL2Gateways(provider)
```

Comment:
Outside the scope of the current PR, but since it touches this code, we might want to rename `fetchL2Gateways` to a better generic name. Infact I would suggest we don;t need the `fetchL2Gateways` file anymore and we can move it inside this file only.

The function can be called `fetchOtherGateways` to better match it's usage.

Reply:
Will be taken up in next PR.